src/api/app/models/project.rb

```diff
@@ -971,7 +968,9 @@ def add_repository_with_targets(repo_name, source_repo, add_target_repos = [], o
 
     # add repository targets
     add_target_repos.each do |repo|
-      trepo.release_targets.create(target_repository: repo, trigger: trigger)
+      unless trepo.release_targets.where(target_repository: repo).exists?
```

Comment:
here it would try to create it also when the trigger as changed leading to a crash, no?